src/openforms/registrations/contrib/zgw_apis/plugin.py

```diff
@@ -38,6 +39,46 @@
 logger = logging.getLogger(__name__)
 
 
+def get_variables_properties_from_submission(
+    submission: Submission, connections: dict[str, Any]
+) -> dict[str, Any]:
+    """
+    Extract the values from the submission and map onto the provided properties (eigenshappen).
+    """
+    variables_properties = {}
+
+    # dict of {componentKey: eigenshap} mapping
+    simple_mappings = {conn["component_key"]: conn["eigenshap"] for conn in connections}
+
+    merged_data = dict(
+        submission.submissionvaluevariable_set.filter(
+            key__in=[conn["component_key"] for conn in connections]
+        ).values_list("key", "value")
+    )
+
+    for key, form_value in merged_data.items():
+        if key not in simple_mappings:
+            continue
+
+        eigenshap = simple_mappings.get(key)
+        variables_properties[eigenshap] = form_value
+
+    return variables_properties
+
+
+class MappedVariablePropertySerializer(serializers.Serializer):
+    component_key = serializers.CharField(
+        label=_("Component key"),
+        help_text=_("Key of the Formio.js component to take the value from."),
+    )
+    eigenshap = serializers.CharField(
+        label=_("Property"),
+        help_text=_(
+            "This is the name of the property with which the variable will be connected."
+        ),
+    )
```

Comment:
You should probably also add validators in here to apply the same validations as `naam` in the Eigenschap resource from https://redocly.github.io/redoc/?url=https://raw.githubusercontent.com/VNG-Realisatie/catalogi-api/1.3.1/src/openapi.yaml#tag/eigenschappen/operation/eigenschap_retrieve:

* max length
* possible character set? Check the implementation of Open Zaak to be sure